### ISSUE TITLE
Temporary Directory Location

### ISSUE THREAD

**Issue by Jan200101** [R2NorthstarTools/FlightCore#463]
Mentioned in #458
Currently FlightCore creates a directory in the game installation to use as a temporary environment to work with.

This has little benefit at the moment but with #456 this would have the following benefits:
- Installing Northstar into the game directory would take considerably less time because file moving is cheap.
- (At least on linux) same fs file moves are atomic, meaning we would explicitly know when something could not be installed.
- end-user is less likely to terminate any update process prematurely by exciting the program, since the processes that take the longest and act on files (e.g. extracting) would not interact with the game files.

A downside may be improper permissions, which would prevent us from creating or using a new directory, but that would also prevent us from installing NS.

The alternative would be using the system provided temporary directory which we would not have to manage and is often backed by ram.

**Comment by Jan200101** [comment]
It may even be best to consider mixing the two, downloading the ns or mod archives into the system temp dir and extracting them into a dir in the game folder, since the archive isn't useful beyond extraction, for which a ramfs would be faster.

**Comment by GeckoEidechse** [comment]
Yeah, for downloading Northstar specifically I was planning to download it into RAM and then extract from there to slightly speed up the install process ^^
